tests/unit/charms/lte_core_interface/v0/test_lte_core_interface_provider.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import unittest
+from ipaddress import AddressValueError
+
+import pytest
+from ops import testing
+
+from tests.unit.charms.lte_core_interface.v0.dummy_provider_charm.src.charm import (
+    DummyCoreProviderCharm,
+)
+
+testing.SIMULATE_CAN_CONNECT = True
+
+BASE_CHARM_DIR = "tests.unit.charms.lte_core_interface.v0.dummy_requirer_charm.src.charm.DummyCoreRequirerCharm"  # noqa: E501
+
+
```

Comment:
What happens if the charm author tries to call `set_core_information` when the relation is not created. That's probably a usecase we want to validate

Reply:
On the other hand, the example in the docstring clearly shows how this method should be used. Do you think we should handle it anyway?

Reply:
I think i'd still do it. For the tls interface, even though it was documented, some charm authors tried to set things before the relation was created. It's pretty easy to validate and test.

Reply:
Yeah, I know it's easy, I'm asking more from the convention perspective. 
The way I looked at it was like "Dear developer, this is the interface for the relation and this is the method this interface provides, to handle relation data changes. How you implement the whole system, it's up to you.".

Reply:
Done. I don't know how the discussion will end up, in case we agree to drop it I'll do it. Let me know what you think.